variables.tf

```diff
@@ -232,3 +265,224 @@ variable "tags" {
+  })
+  default     = null
+  description = "Delegation settings for the API Management service."
+}
+
+variable "gateway_disabled" {
+  type        = bool
+  default     = false
+  description = "Disable the gateway in the main region? This is only supported when additional_location is set."
+  nullable    = false
+  validation {
+    condition     = var.gateway_disabled == false || length(var.additional_location) > 0
+    error_message = "Gateway can only be disabled in the main region when at least one additional location is configured."
+  }
+}
+
+variable "hostname_configuration" {
+  type = object({
+    management = optional(list(object({
+      host_name                       = string
+      key_vault_id                    = optional(string, null)
+      certificate                     = optional(string, null)
+      certificate_password            = optional(string, null)
+      negotiate_client_certificate    = optional(bool, false)
+      ssl_keyvault_identity_client_id = optional(string, null)
+    })), [])
+    portal = optional(list(object({
+      host_name                       = string
+      key_vault_id                    = optional(string, null)
+      certificate                     = optional(string, null)
+      certificate_password            = optional(string, null)
+      negotiate_client_certificate    = optional(bool, false)
+      ssl_keyvault_identity_client_id = optional(string, null)
+    })), [])
+    developer_portal = optional(list(object({
+      host_name                       = string
+      key_vault_id                    = optional(string, null)
+      certificate                     = optional(string, null)
+      certificate_password            = optional(string, null)
+      negotiate_client_certificate    = optional(bool, false)
+      ssl_keyvault_identity_client_id = optional(string, null)
+    })), [])
+    proxy = optional(list(object({
+      host_name                       = string
+      default_ssl_binding             = optional(bool, false)
+      key_vault_id                    = optional(string, null)
+      certificate                     = optional(string, null)
+      certificate_password            = optional(string, null)
+      negotiate_client_certificate    = optional(bool, false)
+      ssl_keyvault_identity_client_id = optional(string, null)
+    })), [])
+    scm = optional(list(object({
+      host_name                       = string
+      key_vault_id                    = optional(string, null)
+      certificate                     = optional(string, null)
+      certificate_password            = optional(string, null)
+      negotiate_client_certificate    = optional(bool, false)
+      ssl_keyvault_identity_client_id = optional(string, null)
+    })), [])
+  })
+  default     = null
+  description = "Hostname configuration for the API Management service."
+}
+
+variable "min_api_version" {
+  type        = string
+  default     = null
```

Comment:
should there be a default version of this? I know you and I talked about this previously and there's discrepancy between expectations vs what actually works in Azure

Reply:
ideally yes, but this variable is really something that is really a niche and we should leave it as is: see https://github.com/hashicorp/terraform-provider-azurerm/pull/28965#issuecomment-2814365865